wagtail/wagtailcore/query.py

```diff
@@ -394,8 +415,15 @@ class SpecificIterable(BaseIterable):
         def __iter__(self):
             return specific_iterator(self.queryset)
 
+    class DeferredSpecificIterable(BaseIterable):
+        def __iter__(self):
+            return specific_iterator(self.queryset, defer=True)
+
 else:
     from django.db.models.query import QuerySet
 
     class SpecificQuerySet(QuerySet):
         iterator = specific_iterator
+
+    class DeferredSpecificQuerySet(QuerySet):
+        iterator = functools.partial(specific_iterator, defer=True)
```

Comment:
It looks like a value for the `qs` argument is missed here. I think this is the reason of test failures like [this](https://travis-ci.org/wagtail/wagtail/jobs/200050157#L438) on Django 1.8:

```
TypeError: specific_iterator() missing 1 required positional argument: 'qs'
```